FrontEndReact/src/View/Admin/View/ViewCourses/AdminViewCourses.js

```diff
@@ -15,55 +15,18 @@ class AdminViewCourses extends Component {
           isLoaded: false,
           courses: [],
       }
+      this.handleGetResource.bind(this);
   }
-  componentDidMount() {
+  async handleGetResource() {
     const cookies = new Cookies();
-    const user_id = cookies.get('user_id');
-    const access_token = cookies.get('access_token');
-    const refresh_token = cookies.get('refresh_token');
-    if(access_token && refresh_token && user_id) {
-        fetch(
-          API_URL + `/course?user_id=${user_id}&admin_id=${user_id}`,
-          {
-              headers: {
-                  "Authorization": "Bearer " + access_token
-              }
-          }
-        )
-        .then(res => res.json())
-        .then(
-            (result) => {
-                if(result["success"]) {
-                    this.setState({
-                        isLoaded: true,
-                        courses: result['content']['courses']
-                    })
-                } else {
-                    const msg = result['msg'];
-                    if(msg==="BlackListed") {
-                        cookies.remove('access_token');
-                        cookies.remove('refresh_token');
-                        cookies.remove('user_id');
-                        window.location.reload(false);
-                    } else if(msg==="Token has expired") {
-                        cookies.remove('access_token');
-                    } else {
-                        this.setState({
-                            isLoaded: true,
-                            errorMessage: result["message"]
-                        })
-                    }
-                }
-            },
-            (error) => {
-                console.log("error: ", error);
-                this.setState({
-                    isLoaded: true,
-                    error: error
-                })
-            }
-        )
-    }
+    await genericResourceFetch(
+        `/course?user_id=${cookies.get('user_id')}&admin_id=${cookies.get('user_id')}`,
```

Comment:
Why does the fetch need both admin_id and user_id? I don't see anywhere in the course routes where a user_id is used, but I do see admin_id.

Reply:
@Wrimo At a first glance, it seems including both the admin_id and user_id is redundant because they both contextually contain the same id of the logged in user. However, both are needed. The attribute in the course table, a foreign key of user_id in the User table, was named admin_id. Therefore, we need to pass a url parameter of admin_id to match the name of the attribute in the Course table. The user_id url parameter is passed because the new authentication decorators look for and parse the user_id from the incoming url. The authentication decorators look specifically for the url parameter of user_id, as how all other incoming url parameters will contain. We would need to rename the admin_id attribute in the Course table to user_id. Then, we would need to update the controller, the model, and the loading demo data functions to match the attribute name of user_id. After making the change, we will be able to remove the url parameter of admin_id and only pass the user_id url parameter.

Reply:
Okay thanks for clarifying. 

Would a minimum solution be to change `request.args.get("admin_id")` to `request.get("user_id")` in the routes? I am currently cleaning up the database, so I can go ahead and make that with the only changes on my branch. 

![image](https://github.com/Lunatic-Labs/rubricapp/assets/38633801/28fca22d-8bd9-4d56-9dff-c50244480673)

Reply:
Yes that would work!

Reply:
Okay cool, and LGTM!